### ISSUE TITLE
Property Injection for fixture instead of ctor Injection

### ISSUE THREAD

**Issue by stefc** [xunit/xunit#2134]
Instead of define in every Test class the same constructor pattern for gain access to the fixture it would be really nice if the framework supports also property Injection  for the fixture if the class supports a specific interface

**Comment by SimonCropp** [comment]
from my experience, most DI containers prefer constructor injection over property injection. perhaps xunit should stick with constructor injection?

**Comment by lucasteles** [comment]
> from my experience, most DI containers prefer constructor injection over property injection. perhaps xunit should stick with constructor injection?

Yes, but for xUnit would be good, so when I inherit from a class with an IClassFixture, would not be necessary to reimplement that constructor repassing the arguments

**Comment by bradwilson** [comment]
Closing this as an alternative exists today in v3: `TestContext.Current.GetFixture()` allows you to retrieve a fixture value without depending on DI.
